parsers/nmparser/structs.go

```diff
@@ -31,9 +31,16 @@ type ParametersData struct {
 	Fixed            ParametersResult   `json:"fixed,omitempty"`
 }
 
+// RunHeuristics ...
+// some values are defined as pointers to support tri-state: true, false, nil
 type RunHeuristics struct {
-	HasZeroGradient        bool `json:"has_zero_gradient,omitempty"`
-	MinimizationSuccessful bool `json:"minimization_successful,omitempty"`
+	CovarianceStepOmitted  string `json:"covariance_step_omitted,omitempty"`
```

Comment:
these should not be strings. They should be typed properly. Please refer to implementations such as:

https://gist.github.com/lummie/7f5c237a17853c031a57277371528e87

and 

https://github.com/campoy/jsonenums

Reply:
That is pretty much the implementation that I have, but when the JSON renders, the values are ints. i

Reply:
where is your marshalling and unmarshalling functionality? 

> but when the JSON renders, the values are ints.

where?

Reply:
Not sure I understand the question, it's serialized using the JSON annotations:
CovarianceStepOmitted  string `json:"covariance_step_omitted,omitempty"`

If the type is an enum/iota, it's serialized as an int

Reply:
its only serialized serialized as an int _if you haven't defined an unmarshal method_. Thats why "That is pretty much the implementation that I have" makes no sense. You literally haven't implemented the key functionality for serializing and deserializing. 

Instead you're coercing to a string that has no type stability on it and would mean that we would have to put conditional logic anywhere else that uses those structs to string match rather than refer to the enum directly.

instead of type stable checks like

```
if RunHeuristics.Minimization == StatusTrue {
...
}
```

you'd need

```
RunHeuristics.Minimization == "True"
```

as you're ToString()'ing everything, which defeats the entire purpose.

It should stay as an enum type until serialization, and upon deserialization should properly reflect the type as well. Thus you need Unmarshal/Marshal methods.